web/src/pages/profile.tsx

```diff
@@ -133,19 +163,31 @@ const ProfilePage = (props: Props): ReactElement => {
   const [formationAddressData, setAddressData] =
     useState<FormationAddress>(emptyFormationAddressData);
   const profileAlertRef = useRef<HTMLDivElement>(null);
+  const isUpdatingUrlFromTabRef = useRef<boolean>(false);
+  const lastUrlTabValueRef = useRef<string | null>(null);
+  const currentTabRef = useRef<ProfileTabs | null>(null);
 
   const FEATURE_EMPLOYER_RATES_ENABLED = process.env.FEATURE_EMPLOYER_RATES === "true";
 
   const getInitTab = (): ProfileTabs => {
     if (props.CMS_ONLY_tab) return props.CMS_ONLY_tab;
 
-    const tabValue = router?.query[QUERIES.tab] as string;
+    const tabValue = router?.query?.[QUERIES.tab];
+    if (tabValue) {
+      const initTab = getProfileTabFromQueryValue(tabValue);
+      if (initTab) return initTab;
+    }
 
-    const initTab = profileTabs.includes(tabValue as ProfileTabs)
-      ? (tabValue as ProfileTabs)
-      : profileTabs[0];
+    if (typeof window !== "undefined") {
+      const url = new URL(window.location.href);
+      const urlTabValue = url.searchParams.get(QUERIES.tab);
+      if (urlTabValue) {
+        const initTab = getProfileTabFromQueryValue(urlTabValue);
+        if (initTab) return initTab;
+      }
+    }
```

Comment:
Is this block of logic necessary? It feels somewhat redundant to the things on lines 176-178

IF it is required, is there a way to consolidate these?